### PR TITLE
feat(Manager): Add DelegatedManager System

### PR DESCRIPTION
**Pull request by pblivin0x**
Proposals
+ https://github.com/SetProtocol/STIPS/blob/main/STIPS/STIP-009.md
+ https://github.com/SetProtocol/STIPS/blob/main/STIPS/STIP-011.md

Contracts
+ Core
    + `ManagerCore`
    + `DelegatedManagerFactory`
    + `DelegatedManager`
+ Global Extensions
    + `GlobalBatchTradeExtension`
    + `GlobalClaimExtension`
    + `GlobalIssuanceExtension`
    + `GlobalStreamingFeeSplitExtension`
    + `GlobalTradeExtension`
    + `GlobalWrapExtension`
+ Library
    + `MutualUpgradeV2`

### CODE REVIEW

File: contracts/global-extensions/GlobalTradeExtension.sol [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        address _sendToken,
+        uint256 _sendQuantity,
+        address _receiveToken,
+        uint256 _minReceiveQuantity,
+        bytes memory _data
+    )
+        external
+        onlyOperator(_setToken)
+        onlyAllowedAsset(_setToken, _receiveToken)
+    {
```
Comment (pblivin0x) [inline contracts/global-extensions/GlobalTradeExtension.sol:140]: can add `mutualUpgrade(methodologist, operator)` here to get multi-sig `trade()` functionality

**Review by FlattestWhite — APPROVED** [review]
LGTM - assuming it's a straight copy paste from set-v2-strategies

**Review by edkim — APPROVED** [review]
Lgtm! There's a failing integration test but I don't know if that's even related to this PR.

**Pull request merged**